src/bin/server/attestations/mod.rs

```diff
@@ -0,0 +1,557 @@
+        return Err(eyre::eyre!("Invalid signature"));
+    }
+    let value = message.message.price.value.to_string();
+    let slot_number = message.message.slot_number.to_string();
+    let signature = message.signature.to_string();
+    let pk_string = validator_public_key.to_string();
+
+    // Save price_value_message in DB
+    sqlx::query!(
+        "
+        INSERT INTO price_value_attestations(
+            validator_public_key,
+            value,
+            slot_number,
+            signature
+        )
+        VALUES (
+            ?1,
+            ?2,
+            ?3,
+            ?4
+        );
+        ",
+        pk_string,
+        value,
+        slot_number,
+        signature,
+    )
+    .execute(db_pool)
+    .await?;
+    Ok(())
+}
+
+async fn save_price_interval_attestations(
+    db_pool: &SqlitePool,
+    messages: &Vec<SignedIntervalInclusionMessage>,
+    validator_public_key: &PublicKey,
+) -> eyre::Result<()> {
+    for message in messages {
+        save_price_interval_attestation(db_pool, message, validator_public_key).await?;
+    }
+    Ok(())
+}
+
+async fn save_price_interval_attestation(
+    db_pool: &SqlitePool,
+    message: &SignedIntervalInclusionMessage,
+    validator_public_key: &PublicKey,
+) -> eyre::Result<()> {
+    if !validate_message(validator_public_key, &message.message, &message.signature) {
+        return Err(eyre::eyre!("Invalid signature"));
+    }
+    let value = message.message.value.to_string();
+    let interval_size = message.message.interval_size.to_string();
+    let slot_number = message.message.slot_number.to_string();
+    let signature = message.signature.to_string();
+    let pk_string = validator_public_key.to_string();
+
+    // Save price_value_message in DB
+    sqlx::query!(
+        "
+        INSERT INTO price_interval_attestations(
+            validator_public_key,
+            value,
+            interval_size,
+            slot_number,
+            signature
+        )
+        VALUES (
+            ?1,
+            ?2,
+            ?3,
+            ?4,
+            ?5
+        );
+        ",
+        pk_string,
+        value,
+        interval_size,
+        slot_number,
+        signature,
+    )
+    .execute(db_pool)
+    .await?;
+
+    // TODO: Review if we really want to aggregate every time we receive a new message
+    extend_or_create_aggregate_interval_attestation(db_pool, message, validator_public_key).await?;
+    Ok(())
+}
+
+async fn extend_or_create_aggregate_interval_attestation(
+    db_pool: &SqlitePool,
+    message: &SignedIntervalInclusionMessage,
+    validator_public_key: &PublicKey,
+) -> eyre::Result<()> {
+    let interval_size = message.message.interval_size.to_string();
+    let slot_number = message.message.slot_number.to_string();
+    let value = message.message.value.to_string();
+    let (new_num_validators, mut aggregate_signature, aggregate_public_key) = if let Some(entry) =
```

Comment:
In general, and if you agree in future code, I'd prefer any assignment is one, or at most a couple lines, this one is 21 😅 .

Reply:
This one is certainly hard to read, so I shortened the whole expression by extracting the sql query part. (In fact all of this in line sql is bothering me the most currently in terms of readability but I guess there is no way around it 😅)

In general I wouldn't say that assignments should be at most a couple of lines. 
The fact that whole if / else blocks can be used as expressions and there result assigned to variables is one of the (many) things I like in rust.
Also I think there are cases where it will just be very hard to fulfil that requirement, such as match expressions across many enum variants..

Reply:
See here for the refactor of this method for improved readability:
https://github.com/ultrasoundmoney/oracle-server/pull/2/commits/914233135ea1f2639ae6b957a0ae9c4584ede04d#diff-0f362055d097365fcff3e4e19de9e5a77d6d820cb16924274b9b14e9c99da664L437